### PR TITLE
feat: initial commit of framework copied from VMClarity

### PR DESCRIPTION
**Pull request by Tehsmash**
Copies the controller framework from VMClarity into this new repo and removes any dependency on VMClarity so that its a fully reusable library.

This PR also introduces CI and the other required repo files so that code quality can be maintained.

### CODE REVIEW

**Review by chrisgacsal — APPROVED** [review]
LGTM! :shipit:

**Review by chrisgacsal — APPROVED** [review]

File: .licensei.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+approved = [
+    "mit",
+    "apache-2.0",
+    "bsd-3-clause",
+    "bsd-2-clause",
+    "mpl-2.0",
+    "isc",
+    "cc0-1.0",
+    "unlicense",
```
Comment (ramizpolic) [inline .licensei.toml:9]: would probably be better to explicitly add unlicensed code to ignored list

File: queue.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	processing map[string]struct{}
+
+	// A map to track items which have been scheduled to be queued at a
+	// later date. We keep track of these items to prevent them being
+	// enqueued earlier than their scheduled time through Enqueue.
+	waitingForEnqueue map[string]struct{}
+
+	// A mutex lock which protects the queue from simultaneous reads and
+	// writes ensuring the queue can be used by multiple go routines safely.
+	l sync.Mutex
```
Comment (ramizpolic) [inline queue.go:67]: we could probably benefit from RWMutex here

**Review by ramizpolic — APPROVED** [review]
LGTM!

File: .licensei.toml (hunk 1 above)
Comment (Tehsmash) [reply .licensei.toml:9]: We'll need to look at that across all the repos 👍

File: queue.go (hunk 2 above)
Comment (Tehsmash) [reply queue.go:67]: When developing this we originally tried to use a RWMutex but it didn't work with the queue model here becayse every operation (Enqueue/Dequeue/Requeue) writes something.

**Pull request merged**